### DISCUSSION TITLE
Use CMake to build GNUStep applications

### DISCUSSION THREAD

**Discussion by dblsaiko** [gnustep/libs-base#359] [General]
Hey,

I want to write a test GUI application in GNUStep, however as far as I can tell (at least looking at the way it's packaged on NixOS), it comes with its own Makefile set as the only package config. Is there some way to load that into (preferably) CMake, or maybe Meson, or is it possible to have GNUStep install a pkg-config or CMake package for each of the libraries?

**Comment by hmelder** [comment]
Take a look at https://github.com/hmelder/Objective-C-Cross-Boilerplate for an example meson build configuration. I use gnustep-config to get all necessary compiler flags.

Shipping CMake configuration just like KDE is doing is certainly possible. AFAIK meson integrates everything into their repo.

**Comment by dblsaiko** [comment]
Perfect, thank you! I missed gnustep-config.

**Comment by dblsaiko** [comment]
Well, so I wrote a cmake find module:

https://git.dblsaiko.net/nucom/tree/cmake/FindGNUstep.cmake?h=rt&id=b3561310c29ddf60e3f14320525fb5ed252e9f9c

```
find_package(GNUstep REQUIRED COMPONENTS Base Gui)
target_link_libraries(MyApp PRIVATE GNUstep::Base GNUstep::Gui)
```

and so on

Not perfect (it definitely doesn't handle non-REQUIRED mode correctly) but it works  for a simple test app (which just calls `NSApplicationMain(int, const char **)`

**Comment by ghost** [comment]
Try this: https://github.com/gnustep/libs-base/discussions/609

**Reply by ghost** [reply]
Meson version: https://github.com/gnustep/libs-base/discussions/610

**Reply by dblsaiko** [reply]
I fixed the link to mine since apparently my paste got deleted. It's a lot more comprehensive and uses import targets which makes it a lot nicer. And as I said on the other discussion, Meson has native support.
